### PR TITLE
Hawthorn installation docs

### PR DESCRIPTION
**Pull request by grantgoodmanedX**
## [DOC-3875](https://openedx.atlassian.net/browse/DOC-3875)

Here's an initial draft of the Open edX Installation and Configuration guide for the Hawthorn release. Points of interest:

- The main change is the addition of the Docker installation instructions for Devstack and the removal of the Vagrant instructions.
- The [release page](https://draft-hawthorn-icr-guide.readthedocs.io/en/latest/platform_releases/hawthorn.html) is mostly a cut and paste from Ginkgo. I'll need to update it with Hawthorn content. 

### Date Needed (optional)

If the release date of a feature is known or estimated, provide it to give reviewers guidance on turnaround time.

### Reviewers

Possible roles follow. The PR submitter checks the boxes after each reviewer finishes and gives :+1:. 

- [x] Subject matter expert: @nedbat 
- [ ] Subject matter expert: 
- [ ] Doc team review (sanity check, copy edit, or dev edit?): @edx/doc
- [ ] Product review:
- [ ] Partner support: 
- [ ] PM review: 

FYI: Tag anyone else who might be interested in this PR here.

### Testing

- [x] Ran ./run_tests.sh without warnings or errors

### HTML Version (optional)

- [x] RTD draft: https://draft-hawthorn-icr-guide.readthedocs.io/en/latest/installation/installation_options.html

### Post-review

- [ ] Squash commits

### CODE REVIEW

**Comment by nedbat** [comment]
@mduboseedx @jmbowman Can you two take a quick look at this PR? I'm a little behind getting rc2 out, but I'd like to make enough progress to be able to push Hawthorn docs by Monday. Thanks.

File: en_us/install_operations/source/installation/install_devstack.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+#. Check out a local copy of the ``edx/devstack`` repository from
+   ``https://github.com/edx/devstack``, selecting the branch that you want to
+   work with.
+
+#. Create a ``devstack`` directory and navigate to it in the command prompt.
+
+   .. code-block:: bash
+
+     mkdir devstack
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/install_devstack.rst]: Cloning the repository in the first step creates and populates this directory automatically.

File: en_us/install_operations/source/installation/install_devstack.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+     cd devstack
+
+#. Create a Python virtual environment. For information about how to do this,
+   see `Virtual Environments`_.
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/install_devstack.rst]: I realize this was in the devstack README, but I actually never use a virtualenv on the host machine when using devstack.  The only direct requirement is `docker-compose`, and Docker for Mac puts a working version of that in the user's `PATH` during installation.  Linux users would need to make sure they have installed it, though.

I'm not thrilled about giving different instructions to Linux vs. Mac users, but creating a virtualenv also tends to be a major sticking point for new Python developers.  I guess I'll leave it to your judgement whether to leave this step and the next one as is, or more explicitly cover ensuring that a suitable version of `docker-compose` is installed.

File: en_us/install_operations/source/platform_releases/hawthorn.rst [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+******************************
+What Is the Hawthorn Git Tag?
+******************************
+
+A git tag identifies the version of Open edX code that is the Hawthorn release.
+You can find the most up-to-date git tag for the current Open edX release on
+the `Open edX Releases Wiki page`_.
+
+The following Open edX git repositories have the Hawthorn git tag:
```
Comment (nedbat) [inline en_us/install_operations/source/platform_releases/hawthorn.rst]: I want to get rid of this list.  Instead, let's say, "About two dozen repos are tagged as part of an Open edX release.  Many other repos are installed as dependencies of those repos."

File: en_us/install_operations/source/platform_releases/hawthorn.rst [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+* edx-ui-toolkit
+* notifier
+* ux-pattern-library
+
+*******************************
+Installing the Hawthorn Release
+*******************************
+
+You can install the Open edX Hawthorn release using :ref:`Devstack <Installing
+the Open edX Developer stack>`.
```
Comment (nedbat) [inline en_us/install_operations/source/platform_releases/hawthorn.rst]: Or by using the Native installation instructions.

File: en_us/install_operations/source/installation/install_devstack.rst [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+    .. code-block:: bash
+    
+     make dev.clone
+
+   To customize where the local repositories are found, set the
+   ``DEVSTACK_WORKSPACE`` environment variable.
+
+#. Share the cloned service directories in Docker, using **Docker ->
+   Preferences -> File Sharing** in the Docker menu.
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/install_devstack.rst]: I think this step is only relevant on macOS; Docker doesn't run in a virtual machine on Linux, so it doesn't need this additional mapping layer.

File: en_us/install_operations/source/platform_releases/hawthorn.rst [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+Hawthorn release on a new machine, and move your data and settings to it.
+
+To move and upgrade your Ginkgo data onto a Hawthorn installation, follow these
+steps.
+
+>>>> IS THIS TRUE?
+
+#. Be sure that your Ginkgo installation is on Ginkgo.2.  The Ginkgo.2 release
+   provided required database migrations beyond what was in Ginkgo.2.  The only
+   version of Ginkgo that will upgrade to Hawthorn successfully is Ginkgo.2.
```
Comment (nedbat) [inline en_us/install_operations/source/platform_releases/hawthorn.rst]: This should say, "Be sure that your Ginkgo installation is on the latest open-release/ginkgo.master to be sure that your database is fully migrated and ready for upgrade to Hawthorn."

File: en_us/install_operations/source/platform_releases/hawthorn.rst [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+   command to check that they are empty and drop them:
+
+    .. code-block:: bash
+
+        $ sudo su - -s /bin/bash edxapp
+        edxapp@xyz:~$ . edxapp_env
+        edxapp@xyz:~$ cd edx-platform/
+        edxapp@xyz:~/edx-platform$ python manage.py lms drop_djcelery_tables --settings=aws
+
+#. Run the Hawthorn migration script, which will update your Ginkgo data to be
```
Comment (nedbat) [inline en_us/install_operations/source/platform_releases/hawthorn.rst]: Change this to "Run the Hawthorn migrations,"

File: en_us/install_operations/source/installation/install_devstack.rst [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    .. code-block:: bash
+     
+     make dev.provision
+
+   As an alternative, use the following command to provision using 
+   :ref:`Docker Sync`.
+
+    .. code-block:: bash
+     
+     make dev.sync.provision
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/install_devstack.rst]: I'd recommend dropping all references to Docker Sync and the optional make targets using it.  It was a performance optimization which is now only marginally faster than default operation, and we've had some reliability issues with it.

File: en_us/install_operations/source/platform_releases/hawthorn.rst [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+
+#. Restart all services.
+
+.. _oscar_hawthorn:
+
+=========================
+Upgrading Django Oscar
+=========================
+
+>> ANYTHING ABOUT DJANGO 1.11? The following is leftover from Ginkgo and prob. no longer applies
```
Comment (nedbat) [inline en_us/install_operations/source/platform_releases/hawthorn.rst]: This section doesn't apply, true.  I'm not sure if we need something about Django 1.11.

**Review by nedbat — CHANGES_REQUESTED** [review]

File: en_us/install_operations/source/installation/install_devstack.rst [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+This stops any running Devstack containers, pulls the latest images, and then starts all of the Devstack containers.
+
+======================================
+Set Up Ability to Preview Course Units
+======================================
+
+If you are installing an Open edX virtual machine on a Linux or Mac computer,
+you must configure your installation to use the preview feature in Open edX
+Studio.
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/install_devstack.rst]: I don't think this is still needed in Docker devstack, but you should probably confirm with someone on the EDUCATOR team to make sure.

File: en_us/install_operations/source/installation/install_devstack.rst [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+****************
+Install Devstack
+****************
+
+To install Devstack, follow these steps.
+
+#. Check out a local copy of the ``edx/devstack`` repository from
+   ``https://github.com/edx/devstack``, selecting the branch that you want to
+   work with.
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/install_devstack.rst:34]: Do you want to cover the instructions on running the containers for a named release from https://github.com/edx/devstack/#how-do-i-run-the-images-for-a-named-open-edx-release early in the steps here?  I can see people who want to run Hawthorn's devstack getting frustrated if they pull the wrong version of the images, etc.

File: en_us/install_operations/source/installation/installation_options.rst [hunk 12]
```diff
@@ -5,87 +5,56 @@ Open edX Platform Installation Options
...
 
 This section describes Open edX installation options and the components that
-each option includes. More details about the various options are at the
-`Open edX Installation Options`_ page on the edX wiki.
-
-There are three virtual machine options, which install the Open edX
-software in a virtual Ubuntu machine.  If you prefer, you can install into an
-Ubuntu machine of your own using the Native installation.
-
+each option includes. There are two virtual machine options, which install
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/installation_options.rst]: Docker uses a single virtual machine under the cover on macOS, but not at all on Linux.  Maybe just say "There are two development environment installation options, ..."?

File: en_us/install_operations/source/installation/installation_options.rst [hunk 13]
```diff
@@ -5,87 +5,56 @@ Open edX Platform Installation Options
...
 **********************************
 
-You can install the Open edX developer stack (**devstack**), the Open edX full
-stack (**fullstack**), or the Open edX analytics developer stack (**analytics
-devstack**).
+You can install the Open edX developer stack (**Devstack**) or the Open edX
+analytics developer stack (**Analytics Devstack**).
 
-* Devstack is a Vagrant virtual machine instance designed for local
+* Devstack is a Docker virtual machine instance designed for local
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/installation_options.rst]: Again, not really a virtual machine.  Something like "Devstack is a set of Docker containers designed for..."

File: en_us/install_operations/source/installation/installation_options.rst [hunk 14]
```diff
@@ -5,87 +5,56 @@ Open edX Platform Installation Options
...
   development. For more information, see :ref:`Info Devstack`.
 
-* Fullstack is a Vagrant virtual machine instance designed for installing all
-  Open edX services on a single server in a production-like configuration. For
-  more information, see :ref:`Info Fullstack`.
-
-* Analytics devstack is a modified version of the devstack virtual machine that
+* Analytics Devstack is a modified version of the Devstack virtual machine that
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/installation_options.rst]: Maybe "Analytics Devstack is a variation on Devstack's Docker containers that..."

File: en_us/install_operations/source/installation/installation_options.rst [hunk 15]
```diff
@@ -5,87 +5,56 @@ Open edX Platform Installation Options
...
   allows you to run Open edX Analytics. For more information, see
   :ref:`Info Analytics Devstack`.
 
-The Open edX Ficus release is supported only on the Ubuntu 16.04 operating
-system. You can use the virtual machine installation options to run Fullstack in
-a Vagrant virtual machine running on Linux, Mac OS, or Windows systems, or to
-run Devstack or Analytics Devstack on Linux or Mac OS. See the `Vagrant`_
-downloads page for information about the operating systems and architectures on
-which you can run Vagrant.
+You can run Devstack or Analytics Devstack on Linux or Mac OS. See the
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/installation_options.rst]: I think it's officially written "macOS" these days, having been renamed from "Mac OS X".

File: en_us/install_operations/source/installation/installation_options.rst [hunk 16]
```diff
@@ -5,87 +5,56 @@ Open edX Platform Installation Options
...
 
 =================
 Open edX Devstack
 =================
 
-Devstack is a Vagrant instance designed for local development. Devstack has
-the same system requirements as :ref:`Fullstack <Installing Open edX
-Fullstack>`. This allows you to discover and fix system configuration issues
-early in development.
+Devstack is a deployment of the Open edX platform within a Docker instance
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/installation_options.rst]: "...within a set of Docker containers designed..."

File: en_us/install_operations/source/installation/installation_options.rst [hunk 17]
```diff
@@ -111,115 +80,34 @@ Native Installation
...
+machine in a production-like configuration. Details are at the `Open edX
 Native Installation`_ page on the edX wiki.
 
 
 *******************
 Software Components
 *******************
 
-All installations include the following Open edX components:
+.. How has this changed, if at all?
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/installation_options.rst]: By default, devstack includes: LMS, Studio, Discussion Forums, ORA, Notes, XQueue, Course Discovery, E-Commerce, Credentials, demo course, and Search (re-order as you see fit).  Programs has been removed from all Hawthorn installations.

File: en_us/install_operations/source/installation/installation_prerequisites.rst [hunk 18]
```diff
@@ -4,23 +4,18 @@
...
 following items.
 
 * Basic terminal usage. If you are using a Mac computer, see
-  `Introduction to the Mac OS X Command Line`_. If you are using a Windows
-  computer, see `Windows Command Line Reference`_.
-
-* Vagrant commands. See the `Vagrant Getting Started`_ guide for
-  more information.
-
+  `Introduction to the Mac OS X Command Line`_.
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/installation_prerequisites.rst:18]: Again, the OS has been renamed.

File: en_us/install_operations/source/installation/start_devstack.rst [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+
+   .. code-block:: bash
+
+    make dev.up
+
+   As an alternative, use :ref:`Docker Sync` with the following command.
+
+   .. code-block:: bash
+
+    make dev.sync.up
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/start_devstack.rst]: Again, probably best to omit Docker Sync.

File: en_us/install_operations/source/installation/start_devstack.rst [hunk 20]
```diff
@@ -0,0 +1,187 @@
...
+   * - ``audit@example.com``
+     - A student account that you can use to access the LMS for testing course
+       auditing.
+   * - ``honor@example.com``
+     - A student account that you can use to access the LMS for testing honor
+       code certificates.
+
+The default password for all of these accounts is ``edx``.
+
+.. _Docker Sync:
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/start_devstack.rst]: Yet more Docker Sync stuff.

File: en_us/install_operations/source/installation/troubleshooting_devstack.rst [hunk 21]
```diff
@@ -0,0 +1,15 @@
+.. _Troubleshooting Devstack:
+
+#####################################
+Troubleshooting Devstack Installation
+#####################################
+
```
Comment (jmbowman) [inline en_us/install_operations/source/installation/troubleshooting_devstack.rst]: Do you want to include some of the content from https://github.com/edx/devstack/#troubleshooting-general-tips and https://github.com/edx/devstack/#troubleshooting-common-issues here?

File: en_us/install_operations/source/platform_releases/hawthorn.rst [hunk 22]
```diff
@@ -0,0 +1,196 @@
...
+.. contents::
+ :local:
+ :depth: 1
+
+****************************
+What's Included in Hawthorn
+****************************
+
+The Open edX Hawthorn release contains several new features for learners,
+course teams, and developers. For more information, see the relesae notes.
```
Comment (jmbowman) [inline en_us/install_operations/source/platform_releases/hawthorn.rst]: "relesae" (Did we ever incorporate a spellchecker into the doc build process?  If not, would it be useful?)

File: en_us/links/links.rst [hunk 23]
```diff
@@ -275,6 +275,14 @@
 
 .. _AWS Documentation: http://aws.amazon.com/documentation/
 
+.. _Docker: https://www.docker.com/community-edition
+
+.. _Docker documentation: https://docs.docker.com/
+
+.. _Docker sync installation: https://github.com/EugenMayer/docker-sync/wiki/1.-Installation
```
Comment (jmbowman) [inline en_us/links/links.rst]: If we remove the Docker Sync references, shouldn't need this.

**Review by jmbowman — COMMENTED** [review]
Pretty good start!  A lot of my comments here are due to flaws in the devstack repo's README that we should have fixed a while ago.

File: en_us/install_operations/source/platform_releases/hawthorn.rst (hunk 9 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/platform_releases/hawthorn.rst]: @jmbowman Do you know of anything we'd want to say about Django 1.11?

File: en_us/install_operations/source/platform_releases/hawthorn.rst (hunk 3 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/platform_releases/hawthorn.rst]: Done.

File: en_us/install_operations/source/platform_releases/hawthorn.rst (hunk 4 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/platform_releases/hawthorn.rst]: Done.

File: en_us/install_operations/source/platform_releases/hawthorn.rst (hunk 6 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/platform_releases/hawthorn.rst]: Done.

File: en_us/install_operations/source/platform_releases/hawthorn.rst (hunk 7 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/platform_releases/hawthorn.rst]: Done.

File: en_us/install_operations/source/installation/install_devstack.rst (hunk 1 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/install_devstack.rst]: Done.

File: en_us/install_operations/source/installation/install_devstack.rst (hunk 5 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/install_devstack.rst]: Done.

File: en_us/install_operations/source/installation/install_devstack.rst (hunk 8 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/install_devstack.rst]: Done.

File: en_us/install_operations/source/installation/install_devstack.rst (hunk 11 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/install_devstack.rst:34]: Done.

File: en_us/install_operations/source/installation/installation_options.rst (hunk 12 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/installation_options.rst]: Done.

File: en_us/install_operations/source/installation/installation_options.rst (hunk 13 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/installation_options.rst]: Done.

Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/installation_options.rst]: Done.

File: en_us/install_operations/source/installation/installation_options.rst (hunk 14 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/installation_options.rst]: Done.

File: en_us/install_operations/source/installation/installation_options.rst (hunk 15 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/installation_options.rst]: I feel so out of touch. Done.

File: en_us/install_operations/source/installation/installation_options.rst (hunk 16 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/installation_options.rst]: Done.

File: en_us/install_operations/source/installation/start_devstack.rst (hunk 19 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/start_devstack.rst]: Done.

File: en_us/install_operations/source/installation/start_devstack.rst (hunk 20 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/start_devstack.rst]: Done.

File: en_us/install_operations/source/platform_releases/hawthorn.rst (hunk 22 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/platform_releases/hawthorn.rst]: Done.

File: en_us/links/links.rst (hunk 23 above)
Comment (grantgoodmanedX) [reply en_us/links/links.rst]: Done.

File: en_us/install_operations/source/platform_releases/hawthorn.rst (hunk 9 above)
Comment (jmbowman) [reply en_us/install_operations/source/platform_releases/hawthorn.rst]: Maybe just a brief note along the lines of the following:

> Whereas Django 1.8 has already reached the end of its support window, Django 1.11 will continue to receive security patches and critical bug fixes until April 2020.  It also has faster database migrations, better support for HiDPI screens in the admin interface, and several useful new features to enhance developer productivity (along with dozens of other bug fixes and minor improvements).

File: en_us/install_operations/source/installation/troubleshooting_devstack.rst (hunk 21 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/troubleshooting_devstack.rst]: Done.

File: en_us/install_operations/source/platform_releases/hawthorn.rst (hunk 9 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/platform_releases/hawthorn.rst]: That's good content for the Hawthorn release notes, but I don' t think we need it in this context - users don't need to do anything in the installation process relating to the Django upgrade. Thanks, Jeremy

File: en_us/install_operations/source/installation/install_devstack.rst (hunk 2 above)
Comment (grantgoodmanedX) [reply en_us/install_operations/source/installation/install_devstack.rst]: I've removed it, and added docker-compose in your path in the installation_prequisites.rst page.

**Comment by nedbat** [comment]
@grantgoodmanedX thanks for the extensive responses.  Are there open questions in here that you think should prevent us from pushing this to readthedocs?

**Review by nedbat — APPROVED** [review]

**Comment by grantgoodmanedX** [comment]
@nedbat I think I can go ahead and publish this. I'm working on that now

**Pull request merged**